lib/generate_patients.rb

```diff
@@ -0,0 +1,132 @@
+require('nokogiri')
+require_relative '../app/models/models'
+require('generate_types')
+
+module QDM
+  # PatientGeneration module can be used to generate patients with dataElements that are populated
+  # with every possible attribute that said type supports
+  module PatientGeneration
+    # Generates patient(s) with fully-loaded dataElements if patient_per_type is false then a
+    # single patient will be returned that has every data elemetnt on it
+    def self.generate_exhastive_data_element_patients(patient_per_type = true)
```

Comment:
This function should be 2 separate functions without the `patient_per_type` boolean passed in.  The comments and return logic below are already somewhat separated, so it should be easy to refactor.